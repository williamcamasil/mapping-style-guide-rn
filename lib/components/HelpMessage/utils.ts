import { ColorPaletesType } from '../../tokens';

export const getTextColor = (error?: boolean, disabled?: boolean): ColorPaletesType => {
  if (disabled) return 'neutralGray500';

  if (error) return 'feedbackError400';

  return 'neutralGray600';
};
