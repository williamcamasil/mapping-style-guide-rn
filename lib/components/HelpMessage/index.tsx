import React, { PropsWithChildren } from 'react';
import { TextStyle } from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Text from '../Text';
import { getTextColor } from './utils';

type InputTextHelpPropsType = PropsWithChildren & {
  theme: AppThemeType;
  error?: boolean;
  disabled?: boolean;
  marginLeft?: TextStyle['marginLeft'];
};

const HelpMessage: React.FC<InputTextHelpPropsType> = ({
  theme,
  children,
  error,
  disabled,
  marginLeft,
}) => {

  const helpStyles = useTextStyles(() => [
    {
      marginTop: theme.spacings.sXXS,
      marginLeft,
    },
  ], [marginLeft, theme.spacings.sXXS]);

  if (!children) return null;

  return (
    <Text
      color={getTextColor(error, disabled)}
      style={helpStyles}
      variant="bodySmall"
      lineHeight="small"
    >
      {children}
    </Text>
  );
};

export default withTheme(HelpMessage);
