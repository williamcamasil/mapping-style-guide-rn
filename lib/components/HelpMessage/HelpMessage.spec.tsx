import React from 'react';

import { render } from '@testing-library/react-native';

import HelpMessage from '.';

describe('HelpMessage snapshot', () => {
  it('default', () => {
    const tree = render(
      <HelpMessage
        error={false}
        disabled={false}
      >
        Mensagem de ajuda
      </HelpMessage>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('empty', () => {
    const tree = render(
      <HelpMessage
        error={false}
        disabled={false}
      >
        {null}
      </HelpMessage>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('error', () => {
    const tree = render(
      <HelpMessage
        error
        disabled={false}
      >
        Mensagem de ajuda
      </HelpMessage>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('disabled', () => {
    const tree = render(
      <HelpMessage
        error={false}
        disabled
      >
        Mensagem de ajuda
      </HelpMessage>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
