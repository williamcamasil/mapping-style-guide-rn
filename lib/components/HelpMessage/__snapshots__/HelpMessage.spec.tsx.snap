// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpMessage snapshot default 1`] = `
<Text
  allowFontScaling={false}
  style={
    {
      "color": "#676D79",
      "flexShrink": 1,
      "fontFamily": "Averta CY",
      "fontSize": 14,
      "fontWeight": "normal",
      "letterSpacing": -0.02,
      "lineHeight": 14,
      "marginLeft": undefined,
      "marginTop": 12,
    }
  }
>
  Mensagem de ajuda
</Text>
`;

exports[`HelpMessage snapshot disabled 1`] = `
<Text
  allowFontScaling={false}
  style={
    {
      "color": "#A4A7AF",
      "flexShrink": 1,
      "fontFamily": "Averta CY",
      "fontSize": 14,
      "fontWeight": "normal",
      "letterSpacing": -0.02,
      "lineHeight": 14,
      "marginLeft": undefined,
      "marginTop": 12,
    }
  }
>
  Mensagem de ajuda
</Text>
`;

exports[`HelpMessage snapshot empty 1`] = `null`;

exports[`HelpMessage snapshot error 1`] = `
<Text
  allowFontScaling={false}
  style={
    {
      "color": "#E53930",
      "flexShrink": 1,
      "fontFamily": "Averta CY",
      "fontSize": 14,
      "fontWeight": "normal",
      "letterSpacing": -0.02,
      "lineHeight": 14,
      "marginLeft": undefined,
      "marginTop": 12,
    }
  }
>
  Mensagem de ajuda
</Text>
`;
